From 52636b9c99de074aee4c6a3a2b36124ba4a6b5b2 Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Fri, 11 Nov 2016 21:23:15 +0800
Subject: [PATCH 1137/2241] fix: spi: a3700: correct register address to
 configure pin number

The Armada3700 SPI instruction, address and data pin number setting
do not take effect since they were configured to SPI control register
by mistake.
As the result the instruction and address pin number will use the
default config to be same as data pin number, which does not have issue
in single mode, but has issue in dual/quad mode.
This patch corrects the register to SPI configuration register, and
always set SPI instruction and address to use 1 interface pin.

Change-Id: I91cfb6216492428ebd4431d9f3eee75bed61c3fc
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33674
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 drivers/spi/spi-armada-3700.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-armada-3700.c b/drivers/spi/spi-armada-3700.c
index d8fba14..7417646 100644
--- a/drivers/spi/spi-armada-3700.c
+++ b/drivers/spi/spi-armada-3700.c
@@ -209,10 +209,10 @@ static int a3700_spi_pin_mode_set(struct a3700_spi *a3700_spi)
 	 * Only support single mode: 1x_instr_pin,
 	 * 1x_addr_pin and 1x_data_pin.
 	 */
-	val = spireg_read(a3700_spi, A3700_SPI_IF_CTRL_REG);
+	val = spireg_read(a3700_spi, A3700_SPI_IF_CFG_REG);
 	val &= ~(A3700_SPI_INST_PIN | A3700_SPI_ADDR_PIN);
 	val &= ~(A3700_SPI_DATA_PIN_MASK << A3700_SPI_DATA_PIN_BIT);
-	spireg_write(a3700_spi, A3700_SPI_IF_CTRL_REG, val);
+	spireg_write(a3700_spi, A3700_SPI_IF_CFG_REG, val);
 
 	return 0;
 }
-- 
2.7.4

